speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -1,229 +1,487 @@
-    if (databaseFile.good()) throw std::runtime_error("A file with the name " + databaseFileName + " already exists");
+    // blackbox
+    std::string blackboxFileName;
+    std::string blackboxVersion;
+    std::string blackboxVersionDate;
+    std::string blackboxVersionTime;
+    std::string blackboxName;
 
-    // open database
-    return sqlite3_open(p_databasePath.c_str(), &p_database);
-}
+    READ_INPUT(blackboxFileName);
+    READ_INPUT(blackboxVersionDate);
+    READ_INPUT(blackboxVersionTime);
+    READ_INPUT(blackboxName);
 
-/// @brief Creates a database and stores data from input file into a table
-/// @param p_filePath path and name of input file (from SimulationData\)
-void SQLDatabaseStorage::StoreData(const std::string p_filePath)
-{
-    std::string SimulationDataPath("SimulationData\\");
-    if (!FindFileDirectory(SimulationDataPath, p_filePath)) throw std::exception("Cannot find SimulationData directory");
+    blackboxVersion = blackboxVersionDate + ' ' + blackboxVersionTime;
+
+    m_values = "'" + blackboxFileName + "','" + blackboxVersion + "','" + blackboxName + "'";
+
+    EXECUTE(INSERT_IGNORE_INTO("blackbox", "filename, version, name", m_values))
+    EXECUTE_QUERY("SELECT blackbox_id FROM blackbox WHERE filename = '" + blackboxFileName + "' AND version = '" + blackboxVersion + "'")
+
+    int blackboxId;
+    GET_INT_FROM_RESULTS(blackboxId)
+
+    // environment
+    std::string environmentFileName;
+    std::string environmentVersion;
+    std::string environmentVersionDate;
+    std::string environmentVersionTime;
+    std::string environmentName;
+
+    READ_INPUT(environmentFileName);
+    READ_INPUT(environmentVersionDate);
+    READ_INPUT(environmentVersionTime);
+    READ_INPUT(environmentName);
 
-    std::ifstream input(SimulationDataPath + p_filePath);
+    environmentVersion = environmentVersionDate + ' ' + environmentVersionTime;
 
-    if (!input.good()) throw std::exception("Cannot find/open input file");
+    m_values = "'" + environmentFileName + "','" + environmentVersion + "','" + environmentName + "'";
 
-    std::string reading;
-    std::stringstream sqlStream;
+    EXECUTE(INSERT_IGNORE_INTO("environment", "filename, version, name", m_values))
+    EXECUTE_QUERY("SELECT environment_id FROM environment WHERE filename = '" + environmentFileName + "' AND version = '" + environmentVersion + "'")
 
-    // read first string: ID of participant
-    input >> reading;
-    const std::string tableName = "Simulation_" + reading;
+    int environmentId;
+    GET_INT_FROM_RESULTS(environmentId)
 
-    sqlite3* database;
-    char* zErrMsg = nullptr;
+    // settings
+    std::string interventionMode;
 
-    std::string databasePath;
+    READ_INPUT(interventionMode);
 
-    // open database
-    int rc = OpenDatabase(database, databasePath, SimulationDataPath, tableName);
-    DATABASE_THROW("Could not open database")
+    m_values = "'" + interventionMode + "'";
```

Comment:
If I'm reading this correctly, you're using a member variable as a temporary variable? I'm not sure whether that's a good idea, but it doesn't seem like one.